pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -497,9 +497,9 @@ cdef class _Timestamp(ABCTimestamp):
             # Matching numpy, we cast to the higher resolution. Unlike numpy,
             #  we raise instead of silently overflowing during this casting.
             if self._creso < other._creso:
-                self = (<_Timestamp>self)._as_creso(other._creso, round_ok=False)
+                self = (<_Timestamp>self)._as_creso(other._creso, round_ok=True)
```

Comment:
Similar question here. Generally pretty wary of casting in C / Cython without explicit type checks, though ideally the type system handles for us

Reply:
thats fair, but the cast is present ATM so id like to consider this out of scope